### PR TITLE
RISC-V tests in QEMU

### PR DESCRIPTION
**Pull request by aykevl**
Not everything is supported right now, but these tests should make sure that things that work now won't break in the future.

### CODE REVIEW

**Comment by deadprogram** [comment]
@aykevl perhaps if you rebase against `dev` this PR might pass the Azure build?

**Comment by aykevl** [comment]
Doubt it, it'll need the SiFive version of QEMU.

**Comment by aykevl** [comment]
Okay, this is ready for review.

I've used a few dirty hacks to get this to work and it doesn't quite work yet on Windows for reasons I can't explain but at least this shows where I'm going.

I think the crude hacks could be avoided by using the `virt` machine for RISC-V instead of `sifive_e`. That target isn't as close to the HiFive1 as `sifive_e`, but it looks like it has a mechanism for exiting tests (the so-called SiFive Test Finisher). Any opinion on this? Rather have dirty hacks or a test that looks less like the real hardware?

Note: the SiFive test finisher is included in the SiFive build of QEMU but somehow didn't end up upstream. Unfortunately, using their version would slow down CI significantly.

File: main_test.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ func TestCompiler(t *testing.T) {
 		runPlatTests("cortex-m-qemu", matches, t)
 	})
 
+	if runtime.GOOS != "linux" {
```
Comment (deadprogram) [inline main_test.go]: Does this also need to exclude `freebsd`?

Comment (aykevl) [reply main_test.go]: Heh, good point. I'll simply check for windows or darwin instead, to be conservative.

**Comment by deadprogram** [comment]
What is the state of this branch? It has a merge conflict that needs to be resolved, but otherwise what is needed for it to be merged?

**Comment by aykevl** [comment]
Right now it has a hack to work around the lack of exiting from within QEMU. I think I should add a new `virt` RISC-V target to fix that.

**Comment by aykevl** [comment]
Investigating this it seems like the first commit will be obsoleted by #871. It would be convenient if that PR was merged first.

**Comment by aykevl** [comment]
Replaced in favor of #976.

**Pull request closed without merging**